engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java

```diff
@@ -1242,4 +1242,87 @@ public void shouldCreateCronTimer() {
                 .exists())
         .isTrue();
   }
+
+  @Test
+  public void shouldAvoidTriggeringMultipleTimes() {
+    // given
+    // set start time to be less than current time
+    final ZonedDateTime start =
+        ZonedDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()).minusHours(1);
+    final long now = engine.getClock().getCurrentTimeInMillis();
+    final BpmnModelInstance model =
+        Bpmn.createExecutableProcess("process")
+            .startEvent("start")
+            .timerWithCycle(String.format("R3/%s/PT2S", start))
+            .endEvent("end")
+            .done();
+    final var deployedProcess =
+        engine
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long processDefinitionKey = deployedProcess.getProcessDefinitionKey();
+
+    // when
+    engine.increaseTime(Duration.ofSeconds(2));
+
+    // then
+    final TimerRecordValue timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .getFirst()
+            .getValue();
+    assertThat(timerRecord.getDueDate()).isBetween(now, now + 2000L);
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+                .withElementId("process")
+                .withProcessDefinitionKey(processDefinitionKey)
+                .exists())
+        .isTrue();
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.CREATED)
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(2)
+                .count())
+        .isEqualTo(2);
+
+    // when
+    engine.increaseTime(Duration.ofSeconds(2));
+
+    // then
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(2)
+                .count())
+        .isEqualTo(2);
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+                .withElementId("process")
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(2)
+                .count())
+        .isEqualTo(2);
+
+    // when
+    engine.increaseTime(Duration.ofSeconds(2));
+
+    // then
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(3)
+                .count())
+        .isEqualTo(3);
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+                .withElementId("process")
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(3)
+                .count())
+        .isEqualTo(3);
```

Comment:
❌ This doesn't really verify the original case. 

But I explained it wrong. Sorry, I didn't fully grasp the topic yet. 🙇 

> if a repeating timer is defined with a start time in the past, then it should trigger only once and reschedule the following triggers in the future

I should have said: if a repeating timer is defined, the engine is stopped for some time during which the timer is supposed to trigger multiple times, and the engine is restarted, then it should trigger only once and reschedule the next timer in the future.

I tried to check this, and noticed that the behavior is different:
- it triggers once for the past due date
- ❌ then once more using the current time as the due date
- and then rescheduled the next timer in the future

I understand that this is somewhat more difficult to write a test for. Let me know if you need a hand. You'll need to:
- deploy a process with a start event (like you do already, but it might be better to schedule it in the future, so we control better when it triggers, e.g. 30 minutes in the future, triggering every 10 minutes)
- stop the engine
- move the clock forward a lot (e.g. an hour)
- reset the RecordingExporter (after a restart the RecordingExporter sees all previous records a second time)
- start the engine again
- verify that there are two timers CREATED:
  - one has a due date in the past
  - one has a due date in the future

Reply:
Thanks for the clarification. ❤️ I will try it.